kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/S3SinkConnectorNetworkIT.java

```diff
@@ -0,0 +1,334 @@
+
+    // assert files in bucket remains same as earlier since the connection is unavailable.
+    assertTrue(fileNamesValid(TEST_BUCKET_NAME, expectedFilenames));
+  }
+
+  @Test
+  @Ignore
+  public void testWithNetworkInterruption() throws Throwable {
+    /*
+     A small value is used to create enough request that the pumba container can cause network
+     interruptions
+    */
+    int flushSize = 3;
+
+    setupSquidProxy();
+    startPumbaPauseContainer();
+    // create topics in Kafka
+    KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+    // send records to kafka
+    sendRecordsToKafka(NUM_RECORDS_PRODUCED);
+
+    props = getConnectorProps();
+    props.put(FLUSH_SIZE_CONFIG, Integer.toString(flushSize));
+
+    // start a sink connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+    // wait for tasks to spin up
+    int minimumNumTasks = Math.min(KAFKA_TOPICS.size(), TASKS_MAX);
+
+    waitForConnectorToStart(CONNECTOR_NAME, minimumNumTasks);
+    int expectedFileCount = NUM_RECORDS_PRODUCED / flushSize;
+    waitForFilesInBucket(TEST_BUCKET_NAME, expectedFileCount);
+    pumbaPauseContainer.close();
+    // assert files in bucket
+    List<String> expectedFilenames = getExpectedFilenames(TEST_TOPIC_NAME, EXPECTED_PARTITION,
+        flushSize, NUM_RECORDS_PRODUCED, EXPECTED_FILE_EXTENSION);
+    assertTrue(fileNamesValid(TEST_BUCKET_NAME, expectedFilenames));
+    shutdownSquidProxy();
+  }
+
+  private void addReadWritePolicyToBucket(String bucketName) {
+    Statement allowRestrictedWriteStatement = new Statement(Statement.Effect.Allow)
+        .withPrincipals(new Principal(System.getenv("SECONDARY_USER_ACCOUNT_ID")))
+        .withActions(S3Actions.GetObject, S3Actions.PutObject)
+        .withResources(new S3ObjectResource(bucketName, "*"));
+    Policy policy = new Policy().withStatements(allowRestrictedWriteStatement);
+    S3Client.setBucketPolicy(bucketName, policy.toJson());
+  }
+
+  private void alterReadWritePolicyOfBucket(String bucketName) {
+    Statement allowRestrictedWriteStatement = new Statement(Statement.Effect.Allow)
+        .withPrincipals(new Principal(System.getenv("SECONDARY_USER_ACCOUNT_ID")))
+        .withActions(S3Actions.GetObject)
+        .withResources(new S3ObjectResource(bucketName, "*"));
+    Policy policy = new Policy().withStatements(allowRestrictedWriteStatement);
+    S3Client.setBucketPolicy(bucketName, policy.toJson());
+  }
+
+  private void sendRecordsToKafka(int numRecords) {
+    // Send records to Kafka
+    for (int i = 0; i < numRecords; i++) {
+      String kafkaTopic = KAFKA_TOPICS.get(i % KAFKA_TOPICS.size());
+      String kafkaKey = "simple-key-" + i;
+      String kafkaValue = "simple-message-" + i;
+      log.debug("Sending message {} with topic {} to Kafka broker {}", kafkaTopic, kafkaValue);
+      connect.kafka().produce(kafkaTopic, kafkaKey, kafkaValue);
+    }
+  }
+
+  private Map<String, String> getConnectorProps() {
```

Comment:
Let's follow the approach in the `S3SinkConnectorIT` class and use the `props` global, and call this method only in the `@Before`